clients/lsp-rust.el

```diff
@@ -1340,12 +1342,21 @@ meaning."
                    (pos (lsp--position-to-point position))
                    (overlay (make-overlay pos pos nil 'front-advance 'end-advance)))
              (let ((concat-label
-                    (string-join
-                     (mapcar
-                      (lambda (label)
-                        (when (lsp-structure-p label)
-                          (gethash "value" label "")))
-                      label))))
+                    (cl-typecase label
+                      (vector
+                       (string-join
+                        (mapcar
+                         (lambda (label)
+                           (when (lsp-structure-p label)
+                             (gethash "value" label "")))
+                         label)))
+                      (string
+                       label)
+                      (t
```

Comment:
this `t` can be replaced with `(not lsp-rust-analyzer-already-war...)`

Reply:
No, because this `t` is a type, not an expression.  Replacing it with that will cause `Unknown type lsp-rust-analyzer-already-warned-about-inlay-hint-type` if neither of the first two branches match.

Reply:
Yeah, you are right.